src/emcpy/plots/domain.py

```diff
@@ -0,0 +1,108 @@
+    lon_formatter = LongitudeFormatter(zero_direction_label=False)
+    lat_formatter = LatitudeFormatter()
+    ax.xaxis.set_major_formatter(lon_formatter)
+    ax.yaxis.set_major_formatter(lat_formatter)
+
+    return ax
+
+
+def _north_america(fig):
+    """
+    Return ax on CONUS domain.
+    """
+    ax = fig.add_subplot(1, 1, 1,
+                         projection=ccrs.PlateCarree(central_longitude=0))
+    ax.add_feature(cfeature.GSHHSFeature(scale='auto'))
+    ax.set_extent([-170, -50, 7.5, 75])
+    ax.set_xticks([-170, -150, -130, -110, -90, -70, -50],
+                  crs=ccrs.PlateCarree())
+    ax.set_yticks([10, 30, 50, 70], crs=ccrs.PlateCarree())
+    lon_formatter = LongitudeFormatter(zero_direction_label=False)
+    lat_formatter = LatitudeFormatter()
+    ax.xaxis.set_major_formatter(lon_formatter)
+    ax.yaxis.set_major_formatter(lat_formatter)
+
+    return ax
+
+
+def _conus(fig):
+    """
+    Return ax on CONUS domain.
+    """
+    ax = fig.add_subplot(1, 1, 1,
+                         projection=ccrs.PlateCarree(central_longitude=0))
+    ax.add_feature(cfeature.GSHHSFeature(scale='auto'))
+    ax.set_extent([-130, -60, 10, 60])
+    ax.set_xticks([-130, -120, -110, -100, -90, -80, -70, -60],
+                  crs=ccrs.PlateCarree())
+    ax.set_yticks([10, 20, 30, 40, 50, 60], crs=ccrs.PlateCarree())
+    lon_formatter = LongitudeFormatter(zero_direction_label=False)
+    lat_formatter = LatitudeFormatter()
+    ax.xaxis.set_major_formatter(lon_formatter)
+    ax.yaxis.set_major_formatter(lat_formatter)
+
+    return ax
+
+
+def _europe(fig):
+    """
+    Returns ax on Europe domain.
+    """
+    ax = fig.add_subplot(1, 1, 1,
+                         projection=ccrs.PlateCarree(central_longitude=0))
+    ax.add_feature(cfeature.GSHHSFeature(scale='auto'))
+    ax.set_extent([-12.5, 40, 30, 70])
+    ax.set_xticks([-10, 0, 10, 20, 30, 40],
+                  crs=ccrs.PlateCarree())
+    ax.set_yticks([30, 40, 50, 60, 70], crs=ccrs.PlateCarree())
+    lon_formatter = LongitudeFormatter(zero_direction_label=False)
+    lat_formatter = LatitudeFormatter()
+    ax.xaxis.set_major_formatter(lon_formatter)
+    ax.yaxis.set_major_formatter(lat_formatter)
+
+    return ax
+
+
+def get_domain(fig, domain):
+    """
+    Factory function to grab correct ax based on inputted domain.
+    Args:
+        fig : matplotlib figure object
+        domain : (str) domain to be plotted on map
+
+    Returns:
+        ax : axis used to plot map on a figure
+    """
+    domain = domain.lower()
+
+    map_domains = {
+        "global": _global,
+        "conus": _conus,
+        "north america": _north_america,
+        "europe": _europe
+    }
+
+    try:
+        return map_domains[domain](fig)
+    except KeyError:
+        raise TypeError(f'{domain} is not a valid domain. Current ' +
+                        'domains supported are:\n' +
+                        'global | conus | north america | europe')
```

Comment:
Print map_domains.keys()? Does that work? So that no need to add it here twice if adding a new domain.

Reply:
Yes! Good catch. Something like this 
```
raise TypeError(f'{domain} is not a valid domain. Current ' +
                         'domains supported are:\n' +
                         f'{[d for d in map_domains.keys()]}')
```
Prints:
TypeError: asia is not a valid domain. Current domains supported are:
['global', 'conus', 'north america', 'europe']

Reply:
```
raise TypeError(f'{domain} is not a valid domain. Current ' + 
                 'domains supported are:\n' + 
                f"{', '.join(map_domains.keys())}")
```

Reply:
Rahul is right, this is the better approach here.